### PR TITLE
introduce global modal with context

### LINKED ISSUE #5964: Make modal component accessible globally

### PR DESCRIPTION
**Pull request by matborowczyk**
# Description

PR will be set to final after merge of #5966 and migrating modals introduced there.

closes #5964 

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [ ] Changelog entry
- [x] Code is clear and sufficiently documented
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      method: "DELETE",
+      headers: createHeaders(env),
+    });
+
+    await handleErrors(response);
+  };
+
+  return useMutation({
+    mutationFn: deleteOrder,
+    mutationKey: ["removeVersion"],
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts:44]: Here too, I'd add in the key that it is about the desired state

File: src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    onSuccess: () => {
+      // Refetch the desired state queries to update the list
+      client.invalidateQueries({
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts:47]: why are you refetching both?

File: src/Data/Managers/V2/GetDesiredStates/getUrl.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
+import moment from "moment-timezone";
+import qs from "qs";
+import { Query, RangeOperator } from "@/Core";
+
+/**
+ * Constructs the URL for fetching desired states based on the provided query parameters.
+ * @note This function is imported from V1 of Query Manager of the same url query manager, as filters are out of scope of the query manager upgrade/migration
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/GetDesiredStates/getUrl.ts:7]: Can you make a ticket to update the getUrl methods to works with React Query, and link it in the comment?

File: src/Data/Managers/V2/GetDesiredStates/useGetDesiredStates.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import { CurrentPage } from "@/Data/Common/UrlState/useUrlStateWithCurrentPage";
+import {
+  DesiredStateVersion,
+  DesiredStateVersionStatus,
+} from "@/Slices/DesiredState/Core/Domain";
+import { PrimaryBaseUrlManager } from "@/UI";
+import { useFetchHelpers } from "../helpers";
+import { getUrl } from "./getUrl";
+
+export interface Filter {
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/GetDesiredStates/useGetDesiredStates.ts:12]: docstrings

File: src/Data/Managers/V2/PromoteVersion/usePromoteVersion.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  const baseUrlManager = new PrimaryBaseUrlManager(
+    globalThis.location.origin,
+    globalThis.location.pathname,
+  );
+  const baseUrl = baseUrlManager.getBaseUrl(process.env.API_BASEURL);
+
+  /**
+   * Sends a request to promote a version of desired state
+   * @param {string} version - the stringified version of desired state.
+   * @throws {Error} If the response is not successful, an error with the error message is thrown.
```
Comment (LukasStordeur) [inline src/Data/Managers/V2/PromoteVersion/usePromoteVersion.ts:31]: Can you add this in the other places too? I notice some fetchers in the other queries don't have docstrings, or that they are not aligned.

File: src/Slices/DesiredState/UI/Components/PromoteAction.tsx [hunk 6]
```diff
@@ -12,31 +13,23 @@ interface Props {
 }
 
 export const PromoteAction: React.FC<Props> = ({ version, isDisabled }) => {
```
Comment (LukasStordeur) [inline src/Slices/DesiredState/UI/Components/PromoteAction.tsx:15]: docstrings

File: src/Slices/DesiredState/UI/Page.tsx [hunk 7]
```diff
@@ -34,76 +41,106 @@ export const Page: React.FC = () => {
...
   });
 
-  function setDeleteModal(version: ParsedNumber, modalState: boolean) {
-    setIsModalOpened(modalState);
-    setVersionToDelete(modalState ? version : 0);
+  const { data, refetch, isError, error, isSuccess } = useGetDesiredStates(
+    envId,
+  ).useContinuous(pageSize, filter, currentPage);
+
+  function setDeleteModal(version: ParsedNumber) {
```
Comment (LukasStordeur) [inline src/Slices/DesiredState/UI/Page.tsx:53]: why did you need the function key word instead of declaring it as a const?

Also, docstrings

File: src/Slices/DesiredState/UI/Page.tsx [hunk 8]
```diff
@@ -34,76 +41,106 @@ export const Page: React.FC = () => {
...
-  );
+            }
+          />
+          <ToastAlert
+            data-testid="ToastAlert"
+            title={words("desiredState.actions.promote.failed")}
+            message={errorMessage}
+            setMessage={setErrorMessage}
+          />
+          {data.data.length <= 0 ? (
```
Comment (LukasStordeur) [inline src/Slices/DesiredState/UI/Page.tsx:129]: How come you end up with data.data?

File: src/Slices/Login/UI/LoginForm/LoginForm.tsx [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+    }
+  }, [isSuccess, navigate, data, authHelper]);
+
+  return (
+    <Form className="loginForm" onSubmit={handleSubmit}>
+      {isError && error && (
+        <FormHelperText>
+          <HelperText>
+            <HelperTextItem
+              variant={isError ? "error" : "default"}
```
Comment (LukasStordeur) [inline src/Slices/Login/UI/LoginForm/LoginForm.tsx:101]: isError will always be true in this case, considering line 97

File: src/Slices/Login/UI/index.ts [hunk 10]
```diff
@@ -0,0 +1 @@
+export * from "./LoginForm";
```
Comment (LukasStordeur) [inline src/Slices/Login/UI/index.ts:1]: Why do we need to export this twice? There are two index files with the same export statement

File: src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx [hunk 11]
```diff
@@ -43,23 +44,23 @@ export const DeleteButton: React.FunctionComponent<DeleteProps> = ({
...
-          onCancel={() => setIsOpen(false)}
-        />
-      </Modal>
       <Button
         variant="secondary"
         isDanger
-        onClick={() => setIsOpen(true)}
+        onClick={() => {
+          triggerModal({
+            title: "Delete Callback",
```
Comment (LukasStordeur) [inline src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx:52]: use the words dict for displayed values

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.tsx [hunk 12]
```diff
@@ -32,13 +30,32 @@ export const DestroyAction: React.FC<Props> = ({
...
   const onSubmit = async () => {
-    setIsOpen(false);
+    closeModal();
     const result = await trigger(refetch);
 
     if (Maybe.isSome(result)) {
       setErrorMessage(result.value);
     }
   };
+  const openModal = () => {
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.tsx:40]: be consistent with the docstrings

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/SetStateSection/SetStateSection.tsx [hunk 13]
```diff
@@ -53,6 +50,32 @@ export const SetStateSection: React.FunctionComponent<Props> = ({
...
+          key="confirm"
+          variant="primary"
+          data-testid={`${id}-state-modal-confirm`}
+          onClick={() => onSubmit(targetState)}
+        >
+          {words("yes")}
+        </Button>,
+        <Button
+          key="cancel"
+          variant="link"
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/Components/RowActionsMenu/SetStateSection/SetStateSection.tsx:70]: link? not secondary?

File: src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx [hunk 14]
```diff
@@ -449,7 +451,7 @@ test("ServiceInventory shows only button to display instance in the composer for
   expect(screen.queryByText("Edit in Composer")).not.toBeInTheDocument();
 });
 
-test("GIVEN ServiceInventory WHEN sorting changes AND we are not on the first page THEN we are sent back to the first page", async () => {
+test.only("GIVEN ServiceInventory WHEN sorting changes AND we are not on the first page THEN we are sent back to the first page", async () => {
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx:454]: remove the only

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+import { FlatEnvironment, Maybe } from "@/Core";
+import { DependencyContext, useNavigateTo } from "@/UI";
+import { ModalContext } from "@/UI/Root/Components/ModalProvider";
+import { words } from "@/UI/words";
+
+interface Props {
+  environment: Pick<FlatEnvironment, "id" | "name">;
+  type: "delete" | "clear";
+}
+export const ConfirmationForm: React.FC<Props> = ({ environment, type }) => {
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:23]: docstrings + newline between interface and component

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 16]
```diff
@@ -0,0 +1,125 @@
...
+  const deleteTrigger = commandResolver.useGetTrigger<"DeleteEnvironment">({
+    kind: "DeleteEnvironment",
+    id: environment.id,
+  });
+  const clearTrigger = commandResolver.useGetTrigger<"ClearEnvironment">({
+    kind: "ClearEnvironment",
+    id: environment.id,
+  });
+
+  async function onConfirm(type: "delete" | "clear") {
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:43]: function?

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 17]
```diff
@@ -0,0 +1,125 @@
...
+        if (validated !== "success" || isBusy) return;
+        onConfirm(type);
+      }}
+    >
+      {errorMessage && (
+        <FormAlert>
+          <Alert
+            data-testid="ErrorAlert"
+            variant="danger"
+            title="Something went wrong"
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:73]: words dict

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 18]
```diff
@@ -0,0 +1,125 @@
...
+              {words("cancel")}
+            </Button>
+          </FlexItem>
+        </Flex>
+      </ActionGroup>
+    </Form>
+  );
+};
+
+const CustomLabel = styled.p`
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:123]: why is this needed?

File: src/UI/Root/Components/ModalProvider/ModalProvider.test.tsx [hunk 19]
```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import { useContext } from "react";
+import { act, render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { ModalContext, ModalProvider } from "./ModalProvider";
+
+const MockedModalUser = () => {
```
Comment (LukasStordeur) [inline src/UI/Root/Components/ModalProvider/ModalProvider.test.tsx:7]: docstrings

File: src/UI/Root/Components/ModalProvider/ModalProvider.tsx [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+  children,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [title, setTitle] = useState("");
+  const [description, setDescription] = useState<React.ReactNode>();
+  const [content, setContent] = useState<React.ReactNode>();
+  const [actions, setAction] = useState<React.ReactNode>([]);
+  const [variant, setVariant] = useState<ModalVariant>(ModalVariant.small);
+  const [iconVariant, setIconVariant] = useState<
+    "success" | "danger" | "warning" | "info" | "custom" | undefined
```
Comment (LukasStordeur) [inline src/UI/Root/Components/ModalProvider/ModalProvider.tsx:35]: maybe use a type for this

**Review by LukasStordeur — COMMENTED** [review]
- update docstrings, I didn't comment on all the places they missed
- be consistent in using function/const and if you need it different, add a comment with the reason
- there are lose strings that should have a words dict

File: src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts (hunk 2 above)
Comment (matborowczyk) [reply src/Data/Managers/V2/DeleteVersion/useDeleteVersion.ts:47]: I changed the word refetch to invalidate, as it doesn't necessarily means that that invalidating will cause refetch, we doing both as we can be in the page that use one or other.

File: src/Slices/DesiredState/UI/Page.tsx (hunk 7 above)
Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.tsx:53]: changed, added docstrings

File: src/Slices/DesiredState/UI/Page.tsx (hunk 8 above)
Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.tsx:129]: That's how result from the endpoint looks like
in the useGetDesiredStates there is interface but basically the endpoints returns:
{
  data: DesiredStateVersion[];
  links: Pagination.Links;
  metadata: Pagination.Metadata;
}

File: src/Slices/Login/UI/index.ts (hunk 10 above)
Comment (matborowczyk) [reply src/Slices/Login/UI/index.ts:1]: I put the component into own folder, I will flatten the structure

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx (hunk 18 above)
Comment (matborowczyk) [reply src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:123]: after checking without whole message in the input label will be bold instead of just name of environment

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/SetStateSection/SetStateSection.tsx (hunk 13 above)
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/SetStateSection/SetStateSection.tsx:70]: I follow what it was already, and modals that don't use confirm action form had link variant

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.tsx (hunk 12 above)
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.tsx:40]: won't Service Inventory won't be replaced by your redesign?

Comment (LukasStordeur) [reply src/Slices/ServiceInventory/UI/Components/RowActionsMenu/DestroyAction/DestroyAction.tsx:42]: Doesn't mean the current code shouldn't be clean and aligned. The redesign will remove the collapsible content, and replace the buttons, yes, but that is for a later step.

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx (hunk 18 above)
Comment (LukasStordeur) [reply src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:149]: Can you add the explanation in as a comment?

File: src/Slices/Login/UI/LoginForm.tsx [hunk 21]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+/**
+ * LoginForm component.
+ * @param {LoginFormProps} props - The component props.
+ * @returns {JSX.Element} The rendered component.
+ */
+export const LoginForm: React.FC<LoginFormProps> = ({
+  submitButtonText,
+  submitButtonLabel = "login-button",
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/Login/UI/LoginForm.tsx:38]: we don't add aria-labels to the dictionary as far as I know

File: src/Slices/DesiredState/UI/Page.test.tsx [hunk 22]
```diff
@@ -24,681 +23,575 @@ import {
...
-  await act(async () => {
-    await apiHelper.resolve(
-      Either.right({
-        ...DesiredStateVersions.response,
-        data: DesiredStateVersions.response.data.slice(0, 3),
+  it("When using the Version filter then the desired state versions within the range selected range should be fetched and shown", async () => {
+    server.use(
+      http.get("/api/v2/desiredstate", async ({ request }) => {
+        if (
+          request.url.split("?")[1] ===
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.test.tsx:370]: I feel like using for example includes() leave a space for some additional parameters at the end which could not be allowed

File: src/Slices/DesiredState/UI/Page.tsx [hunk 23]
```diff
@@ -34,76 +41,111 @@ export const Page: React.FC = () => {
...
 
-  function setDeleteModal(version: ParsedNumber, modalState: boolean) {
-    setIsModalOpened(modalState);
-    setVersionToDelete(modalState ? version : 0);
+  const { data, refetch, isError, error, isSuccess } = useGetDesiredStates(
+    envId,
+  ).useContinuous(pageSize, filter, currentPage);
+
+  /**
+   * callback function that will open a modal to confirm action to delete a version
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/DesiredState/UI/Page.tsx:54]: I personally would not use "callback" to describe useCallback, but I will rename it as it isn't 100% correct anyways

File: src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx [hunk 24]
```diff
@@ -1,33 +1,46 @@
...
 
-interface DeleteProps {
+interface Props {
   callback: Callback;
   service_entity: string;
 }
 
-export const DeleteButton: React.FunctionComponent<DeleteProps> = ({
+/**
+ * DeleteButton component which holds the logic for deleting a desired state version.
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx:16]: right, I thought I was editing the other button

File: src/UI/Root/Components/ModalProvider/ModalProvider.tsx [hunk 25]
```diff
@@ -0,0 +1,142 @@
+import React, { PropsWithChildren, createContext, useState } from "react";
+import { Modal, ModalVariant } from "@patternfly/react-core";
+
+type IconVariant = "success" | "danger" | "warning" | "info" | "custom";
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:4]: I checked that before even creating that in first iteration, as I had `ModalVariant` here but nothing popped out

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:4]: `import { variantIcons } from "@patternfly/react-core";`
 it is available :) we already use it in some places too.

[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:4]: I was searching for IconVariant as they have ModalVariant and thought it would follow the convention

[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:4]: After checking, variantIcons is a object that holds Icons svgs, not type/enum :)

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:4]: right, my bad. I would have expected them to have a status type

File: src/Slices/Login/UI/LoginForm.tsx [hunk 26]
```diff
@@ -0,0 +1,182 @@
...
+    setPassword(value);
+  };
+
+  /**
+   * Handles the submission of the login form.
+   *
+   * This function is responsible for preventing the default form submission behavior and then calling the mutate function with the current username and password.
+   * @param {React.FormEvent<HTMLFormElement> | React.MouseEvent<HTMLButtonElement, MouseEvent>} event - The event that triggered the form submission.
+   *
+   * @returns {void}
```
Comment (LukasStordeur) [inline src/Slices/Login/UI/LoginForm.tsx:86]:
```suggestion
   * @returns {void} A Promise that resolves when the operation is complete.
```

File: src/Slices/Login/UI/LoginForm.tsx [hunk 27]
```diff
@@ -0,0 +1,182 @@
...
+          value={username}
+          onChange={handleUsernameChange}
+        />
+      </FormGroup>
+      <FormGroup
+        label={words("password")}
+        isRequired
+        fieldId="pf-login-password-id"
+      >
+        {
```
Comment (LukasStordeur) [inline src/Slices/Login/UI/LoginForm.tsx:140]: why is this wrapped in brackets? I don't think it's needed

File: src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx [hunk 28]
```diff
@@ -1,33 +1,49 @@
...
 
-export const DeleteButton: React.FunctionComponent<DeleteProps> = ({
+/**
+ * DeleteButton component which holds the logic for deleting a callback.
+ *
+ * @props {Props} props - The props of the component.
+ * @prop callback {Callback} - the callback object
+ * @prop service_entity {string} - the service entity name
+ *
+ * @returns {React.React.FC<Props> }
```
Comment (LukasStordeur) [inline src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx:22]:
```suggestion
 * @returns {React.React.FC<Props>} The rendered Component to delete a callback.
```

File: src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx [hunk 29]
```diff
@@ -1,33 +1,49 @@
...
-  const [isOpen, setIsOpen] = useState(false);
   const [errorMessage, setErrorMessage] = useState("");
 
-  const onSubmit = async () => {
-    setIsOpen(false);
+  /**
+   * submit function that will close a modal and trigger the delete action
+   * if there is an error, it will set the error message
+   *
+   * @returns {Promise<void>}
```
Comment (LukasStordeur) [inline src/Slices/ServiceDetails/UI/Tabs/Callbacks/DeleteButton.tsx:43]:
```suggestion
   * @returns {Promise<void>} A Promise that resolves when the operation is complete.
```

File: src/Slices/ServiceInventory/UI/Components/RowActionsMenu/ForceStateAction/ForceStateAction.tsx [hunk 30]
```diff
@@ -64,20 +66,71 @@ export const ForceStateAction: React.FC<Props> = ({
 
   const isHalted = environmentModifier.useIsHalted();
 
+  /**
+   * Opens a modal with a confirmation buttons.
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/Components/RowActionsMenu/ForceStateAction/ForceStateAction.tsx:70]:
```suggestion
   * Opens a modal with confirmation buttons.
```

File: src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx [hunk 31]
```diff
@@ -498,7 +500,7 @@ test("GIVEN ServiceInventory WHEN sorting changes AND we are not on the first pa
 
   //sort on the second page
   await act(async () => {
-    await userEvent.click(screen.getByRole("button", { name: "State" }));
+    await userEvent.click(screen.getAllByRole("button", { name: "State" })[0]);
```
Comment (LukasStordeur) [inline src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx:503]: How comes there are suddenly multiple buttons with the same aria-label?

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 32]
```diff
@@ -0,0 +1,147 @@
...
+  type: EnvActions;
+}
+
+/**
+ * ConfirmationForm component.
+ * @props {Props} props - The component props.
+ * @prop {FlatEnvironment} environment - An object that represents the environment. It is a subset of the `FlatEnvironment` type, including only the `id` and `name` properties.
+ * @prop {EnvActions} type- The type of action to perform.
+ *
+ * @returns {React.FC <Props>} - The rendered confirmation form.
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:31]:
```suggestion
 * @returns {React.FC<Props>} - The rendered confirmation form.
```

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx [hunk 33]
```diff
@@ -0,0 +1,147 @@
...
+import { words } from "@/UI/words";
+import { EnvActions } from "./Actions";
+
+interface Props {
+  environment: Pick<FlatEnvironment, "id" | "name">;
+  type: EnvActions;
+}
+
+/**
+ * ConfirmationForm component.
```
Comment (LukasStordeur) [inline src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:26]: Confirm what?

File: src/Slices/UserManagement/UI/Components/UserInfoRow.tsx [hunk 34]
```diff
@@ -1,41 +1,58 @@
...
-interface UserInfoRowProps {
+interface Props {
   user: UserInfo;
 }
 
 /**
  * A functional component that renders a row in the user information table.
- * @param user The user information.
+ * @props {Props} props The props of the component.
+ * @prop {UserInfo} user The user information.
```
Comment (LukasStordeur) [inline src/Slices/UserManagement/UI/Components/UserInfoRow.tsx:17]:
```suggestion
 * @prop {UserInfo} user - The user information.
```

File: src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx (hunk 31 above)
Comment (matborowczyk) [reply src/Slices/ServiceInventory/UI/ServiceInventory.test.tsx:503]: During the development I got out of the blue error that there is multiple elements with that name, and the correct one was 2 in the array. but at the end it was failing again, so I thought it got somehow reordered, I don't know what happened here, as I wasn't changing anything there, but it got somehow to back to the original way - changed back to what it was

File: src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx (hunk 33 above)
Comment (matborowczyk) [reply src/Slices/Settings/UI/Tabs/Environment/Components/ConfirmationForm.tsx:26]: added

File: src/UI/Root/Components/ModalProvider/ModalProvider.tsx [hunk 35]
```diff
@@ -0,0 +1,140 @@
...
+  const [iconVariant, setIconVariant] = useState<IconVariant | undefined>(
+    "info",
+  );
+
+  /**
+   * Triggers the modal with the provided properties.
+   *
+   * This function updates the modal's properties and opens the modal.
+   * The properties include the title, description, content, actions, variant, and icon variant.
+   * The description and actions default to `null` and an empty array, respectively, if they are not provided.
```
[the remark this answers is not in the document]
Comment (matborowczyk) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:81]: I wouldn't personally go in more details here, parameters are, I think, clearly defined set to state which is assigned to modal component, in my opinion there is nothing more valuable to add here. renamed @prop to @param

[the remark this answers is not in the document]
Comment (LukasStordeur) [reply src/UI/Root/Components/ModalProvider/ModalProvider.tsx:81]: You're still using `@prop` below.
As for the explanation, I had to explicitly look up on  PF to know what was meant with the actions. Sometimes, it is not always clear whether we are using custom properties/arguments or the original ones from the library. Maybe worth knowing that the footer element in the content will override the action content. Or at least reference that these are the original attributes of PF

**Review by LukasStordeur — APPROVED** [review]
